share_extension/ios/extension_channels.js

```diff
@@ -117,39 +117,48 @@ export default class ExtensionChannels extends PureComponent {
     loadChannels = async () => {
         try {
             const {currentUserId, teamId} = this.props;
+            const usersInDms = [];
             const channelsMap = {};
 
+            // get the channels for the specified team
             const myChannels = await Client4.getMyChannels(teamId);
-            const myPreferences = await Client4.getMyPreferences();
 
-            const usersInDms = myPreferences.filter((pref) => pref.category === Preferences.CATEGORY_DIRECT_CHANNEL_SHOW).map((pref) => pref.name);
+            // filter channels that are direct and group messages
             const dms = myChannels.filter((channel) => {
-                const teammateId = getUserIdFromChannelName(currentUserId, channel.name);
-                return (channel.type === General.DM_CHANNEL && usersInDms.includes(teammateId)) || channel.type === General.GM_CHANNEL;
+                if (channel.type === General.DM_CHANNEL) {
+                    const teammateId = getUserIdFromChannelName(currentUserId, channel.name);
+                    if (!usersInDms.includes(teammateId)) {
+                        usersInDms.push(teammateId);
```

Comment:
I don't know how I feel about having side effects inside a filter since it goes against the functional nature of filter, but I guess it's okay

Reply:
I have the same thought in here. I feel like `dms` and `usersInDms` should have it's own run, could be like (not tested if this will work though),
```
const dms = myChannels.filter((channel) => {
    return channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL;
});

const usersInDms = dms.filter((channel) => {
    return channel.type === General.DM_CHANNEL;
}).map((channel) => {
    return getUserIdFromChannelName(currentUserId, channel.name);
}).reduce((acc, teammateId) => {
    if (!acc.includes(teammateId)) {
        acc.push(teammateId);
    }

    return acc
}, []);
```